Improve secrets management

jmaupetit opened the pull request:
## Purpose

Secrets generation is for now limited to a generic template that handles key:values that can be injected as environment variables to a running pod. We are not able to define other type of secrets than Opaque ones, and we are restricted to one secret per application. We think there is room for improvement.

## Proposal

- [x] allow to define and create secrets template in service templates for an app
- [x] add new templates for `richie`'s secrets
- [x] add new templates for `edxapp`'s secrets

jmaupetit committed:
⚡️(secrets) make secrets management more flexible

Instead of using a limited generic template to create secrets for an
application, we now allow users to define their own secrets templates
for their services: those templates should be called
"secrets.*\.yml\.j2" and stored along with other templates for a
service.

jmaupetit committed:
⚡️(richie) add new application secrets

Secrets are now stored separately in a dedicated template for each
service. Vars are defined in the vault, decrypted on the fly and
injected in the template of the secret. Note that secret name and
versionning should be done using new variables defined for the
application, e.g. {{ app.name }}_{{ app_service }}_secret_name.

jmaupetit committed:
⚡️(edxapp) add new application secrets

We now separate secrets per service, e.g. mongodb, mysql, and edxapp.
In the present work, edxapp CMS + LMS secrets are all set via the
settings.yml file instead of a mix of environment and the
aforementioned file.

sampaccoud commented on apps/edxapp/templates/mongodb/secret.yml.j2:
```diff
@@ -0,0 +1,13 @@
+  namespace: "{{ project_name }}"
+data:
+  MONGODB_ADMIN_PASSWORD: "{{ MONGODB_ADMIN_PASSWORD | default('password') | b64encode }}"
+  MONGODB_USER: "{{ MONGODB_USER | default('fun') | b64encode }}"
```
Should we call it "edxapp_user"?

sampaccoud commented on apps/richie/templates/app/secret.yml.j2:
```diff
@@ -0,0 +1,13 @@
+data:
+  DJANGO_SECRET_KEY: "{{ DJANGO_SECRET_KEY | default('supersecret') | b64encode }}"
+  DJANGO_SENTRY_DSN: "{{ DJANGO_SENTRY_DSN | default('fakedsn') | b64encode }}"
+  POSTGRES_USER: "{{ POSTGRESQL_USER | default('fun') | b64encode }}"
```
ditto

sampaccoud commented on apps/richie/templates/postgresql/secret.yml.j2:
```diff
@@ -0,0 +1,11 @@
+  name: "{{ richie_postgresql_secret_name }}"
+  namespace: "{{ project_name }}"
+data:
+  POSTGRESQL_USER: "{{ POSTGRESQL_USER | default('fun') | b64encode }}"
```
ditto

jmaupetit committed:
🔇(objects) increase level of app objects listing

Display all object templates for an app shouldn't be the default, this
is clearly for at least verbosity level 2.

sampaccoud commented on tasks/get_objects_for_app.yml:
```diff
@@ -46,6 +46,7 @@
 - name: Display OpenShift's builds for this app
```
in commit message: s/should'nt/shouldn't
